tests/helpers/test_script.py

```diff
@@ -1652,17 +1048,20 @@ def wait_started_cb():
         # This should stop first run then start a new run.
 
         wait_started_flag.clear()
-        await script_obj.async_run()
+        hass.async_create_task(script_obj.async_run())
         await asyncio.wait_for(wait_started_flag.wait(), 1)
 
         assert script_obj.is_running
         assert len(events) == 2
         assert events[1].data["value"] == 1
-        assert any(
-            rec.levelname == "INFO"
-            and rec.name == "TEST"
-            and "Restarting" in rec.message
-            for rec in caplog.records
+        assert all(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5164426)